tests/vm_test.go

```diff
@@ -1346,61 +1343,55 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
-					By("Starting the VMI with virtctl")
-					startCommand := clientcmd.NewRepeatableVirtctlCommand(virtctl.COMMAND_START, "--namespace", vm.Namespace, vm.Name)
-					Expect(startCommand()).To(Succeed())
-
-					By("Waiting for VM to be in Starting status")
-					Eventually(ThisVM(vm), 160*time.Second, 1*time.Second).Should(HavePrintableStatus(v1.VirtualMachineStatusStarting))
-
 					By("Waiting for VMI to fail")
 					Eventually(ThisVMIWith(vm.Namespace, vm.Name), 480*time.Second, 1*time.Second).Should(BeInPhase(v1.Failed))
+					vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+					Expect(err).ShouldNot(HaveOccurred())
+
+					getComputeContainerStateRunning := func() (*k8sv1.ContainerStateRunning, error) {
+						launcherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
+						if err != nil {
+							return nil, err
+						}
+						for _, status := range launcherPod.Status.ContainerStatuses {
+							if status.Name == "compute" {
+								return status.State.Running, nil
+							}
+						}
+						return nil, nil
+					}
 
 					// If the annotation v1.KeepLauncherAfterFailureAnnotation is set to true, the containerStatus of the
 					// compute container of the virt-launcher pod is kept in the running state.
 					// If the annotation v1.KeepLauncherAfterFailureAnnotation is set to false or not set, the virt-launcher pod will become failed.
 					By("Verify that the virt-launcher pod or its container is in the expected state")
-					vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					launcherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vm.Namespace)
-					Expect(err).ToNot(HaveOccurred())
-
 					if toKeep, _ := strconv.ParseBool(keepLauncher); toKeep {
-						Consistently(func() bool {
-							for _, status := range launcherPod.Status.ContainerStatuses {
-								if status.Name == "compute" && status.State.Running != nil {
-									return true
-								}
-							}
-							return false
-						}, 10*time.Second, 1*time.Second).Should(BeTrue())
+						Consistently(func() *k8sv1.ContainerStateRunning {
+							computeContainerStateRunning, err := getComputeContainerStateRunning()
+							Expect(err).ToNot(HaveOccurred())
+							return computeContainerStateRunning
+						}).WithTimeout(10 * time.Second).WithPolling(1 * time.Second).ShouldNot(BeNil())
 					} else {
-						Eventually(launcherPod.Status.Phase).
-							Within(160 * time.Second).
-							WithPolling(time.Second).
-							Should(Equal(k8sv1.PodFailed))
+						Eventually(func() *k8sv1.ContainerStateRunning {
+							computeContainerStateRunning, err := getComputeContainerStateRunning()
+							Expect(err).ToNot(HaveOccurred())
+							return computeContainerStateRunning
+						}).WithTimeout(100 * time.Second).WithPolling(1 * time.Second).Should(BeNil())
```

Comment:
Let's also assert the Pod as whole is `Failed`

Reply:
Done